pkg/storage/odf_test.go

```diff
@@ -0,0 +1,401 @@
+	}{
+		{
+			testStorageSystem: buildValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     nil,
+		},
+		{
+			testStorageSystem: buildInValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     fmt.Errorf("storageSystem 'name' cannot be empty"),
+		},
+		{
+			testStorageSystem: buildValidStorageSystemBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError: fmt.Errorf("storagesystems.odf.openshift.io \"ocs-storagecluster-storagesystem\" " +
+				"not found"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		storageSystemObj, err := testCase.testStorageSystem.Get()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, storageSystemObj.Name, testCase.testStorageSystem.Definition.Name)
+			assert.Equal(t, storageSystemObj.Namespace, testCase.testStorageSystem.Definition.Namespace)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestStorageSystemCreate(t *testing.T) {
+	testCases := []struct {
+		testStorageSystem *StorageSystemBuilder
+		expectedError     string
+	}{
+		{
+			testStorageSystem: buildValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     "",
+		},
+		{
+			testStorageSystem: buildInValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     "storageSystem 'name' cannot be empty",
+		},
+		{
+			testStorageSystem: buildValidStorageSystemBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:     "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testStorageSystemBuilder, err := testCase.testStorageSystem.Create()
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testStorageSystemBuilder.Definition.Name, testStorageSystemBuilder.Object.Name)
+			assert.Equal(t, testStorageSystemBuilder.Definition.Namespace, testStorageSystemBuilder.Object.Namespace)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		}
+	}
+}
+
+func TestStorageSystemDelete(t *testing.T) {
+	testCases := []struct {
+		testStorageSystem *StorageSystemBuilder
+		expectedError     error
+	}{
+		{
+			testStorageSystem: buildValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     nil,
+		},
+		{
+			testStorageSystem: buildInValidStorageSystemBuilder(buildStorageSystemClientWithDummyObject()),
+			expectedError:     fmt.Errorf("storageSystem 'name' cannot be empty"),
+		},
+		{
+			testStorageSystem: buildValidStorageSystemBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:     nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testStorageSystem.Delete()
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testStorageSystem.Object)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
```

Comment:
Consolidate the error checks above  line 273

Reply:
imposible

Reply:
Why is this impossible? Isn't it the same as the other cases I mentioned?

Reply:
I cannot parse error message in this case, only if I compare the whole error.
In this specific case it is working, but there are many it will fail